Formula/s/slackdump.rb

```diff
@@ -0,0 +1,26 @@
+class Slackdump < Formula
+  desc "Export Slack data without admin privileges"
+  homepage "https://github.com/rusq/slackdump"
+  url "https://github.com/rusq/slackdump/archive/refs/tags/v2.5.10.tar.gz"
+  sha256 "813c282b3fe64f9c75ae911af85a8524ccb241b0e73a45d23a4f222622531d43"
+  license "GPL-3.0-only"
+  head "https://github.com/rusq/slackdump.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.date=#{time.iso8601}
+      -X main.commit=Homebrew
```

Comment:
Why different ldflags?
https://github.com/rusq/slackdump/blob/cafaee525acf22e20a33b3ec7336912fbf88c10d/Makefile#L11
`LDFLAGS="-s -w -X 'main.build=$(BUILD)' -X 'main.buildYear=$(BUILD_YEAR)'"`

Reply:
Could it be that their own `Makefile` is outdated?
https://github.com/rusq/slackdump/blob/master/cmd/slackdump/version.go

Reply:
Would be good to keep this as an actual commit hash.

Reply:
I think so too, but I know @chenrui333 has been working to remove all commit hashes from Core. So can we decide on a policy for Core then?

Reply:
> I know @chenrui333 has been working to remove all commit hashes from Core

Why?

Reply:
it is not very useful for release build and it has to be git checkout rather than tarball.

Reply:
For some formulae it makes sense to use a stand-in for the commit hash.

For example,
```
❯ fzf --version
0.54.3 (brew)
```
Putting `brew` there was actually an [upstream contribution](https://github.com/Homebrew/homebrew-core/commit/01cc4648bb2ad6474db92f639ef24b20e86beffc), and it probably makes triaging bug reports easier for them.

On the other hand, I don't think it quite works here:
```
❯ ./slackdump -V
Slackdump 2.5.10 (commit: 0f0812a08f5ed3d71f9b03ac850b29439e3760f8) built on: 2024-08-15T11:32:47Z
2.5.10
```

More generally I think that as long as the version output doesn't end up saying something like `commit: Homebrew` and upstream/users don't complain, we should probably avoid Git checkouts.

Reply:
I think we should use whatever format of checkout produces the most information in the build by default. In cases like `fzf` if upstream specifically doesn't want a commit hash: cool, we can oblige there, but generally we should just do a Git checkout.

The overhead in our CI isn't that large, it doesn't apply to most users and it's more like how upstream expects their software to be built.

Reply:
I still think from packaging perspective, it does make sense to just use tarball rather than git checkout. If we have concerns with the version output we can also leave it empty though (based on my experience, upstream is generally fine with the homebrew or brew in there, like @carlocab said it would be helpful for bug reporting).

Reply:
> I still think from packaging perspective, it does make sense to just use tarball rather than git checkout.

Can you elaborate on why? Is the `commit` in here never useful information for us or upstream?

Reply:
also, if CI overhead is not big concern (which would be surfaced whenever we do the major go bumps), I am also okay switching back to git checkout. (IIRC the first tarball vs git checkout debate was on the CI performance) cc @branchvincent

Reply:
> Why different ldflags?
> https://github.com/rusq/slackdump/blob/cafaee525acf22e20a33b3ec7336912fbf88c10d/Makefile#L11
> `LDFLAGS="-s -w -X 'main.build=$(BUILD)' -X 'main.buildYear=$(BUILD_YEAR)'"`

58b602e4903d5d8df6e8a19a54613e79ecd246f0

Reply:
> > I still think from packaging perspective, it does make sense to just use tarball rather than git checkout.
> 
> Can you elaborate on why? Is the `commit` in here never useful information for us or upstream?

Yeah, I think it is never useful, for upstream or go community, it is mostly used for debugging dev builds, not prod release builds.

Reply:
I can see the argument against it in cases like this where we also hardcoded the version. If we don't hardcoded the tag version: the specific commit seems very useful. Even then, though, we could potentially use a tag. The main thing we should make sure we never do is set the version/tag/commit to just `Homebrew` or `brew` if it means version information is not also being stored in the binary.

Reply:
> The main thing we should make sure we never do is set the version/tag/commit to just `Homebrew` or `brew` if it means version information is not also being stored in the binary.

Agreed. The good news is that I don't think this is the case for any formulae. Usually when we fail to store version information in the binary, it's because we omitted passing the right `-X` flags entirely.

Reply:
yeah, sounds like we should run another rfc chat about the decision :) (I guess we can just move forward with this PR merge?)

Reply:
>>  The main thing we should make sure we never do is set the version/tag/commit to just Homebrew or brew if it means version information is not also being stored in the binary.

> Agreed. The good news is that I don't think this is the case for any formulae. Usually when we fail to store version information in the binary, it's because we omitted passing the right -X flags entirely.

yeah, all the buildDate, buildCommit was for debugging before, now we have toolchain, I think there might be an need to have another ldflag to manifest that (I know golangci-lint has issue before, and it looks like they have included the build go version in the version output)

Reply:
> The good news is that I don't think this is the case for any formulae.

If this is the case: that's fine, we are fine to continue using tarballs here and move forward with both the PR merge and don't need a RFC.

Reply:
Sounds good, thanks for the chat, let's move forward.